apis/v1alpha1/webhook.go

```diff
@@ -83,6 +84,9 @@ func (r *CrdbCluster) Default() {
 func (r *CrdbCluster) ValidateCreate() error {
 	webhookLog.Info("validate create", "name", r.Name)
 
+	if r.Spec.CockroachDBVersion == "" && (r.Spec.Image == nil || r.Spec.Image.Name == "") {
```

Comment:
Should we do the same check for updates?

Reply:
Also, I think we're missing a test case for this

Reply:
Added for updates as well and added another check if both is provided by the user then also give warning. Also, added the tests for the same

Reply:
these checks look a little complex and are duplicated across `ValidateCreate` and `ValidateUpdate` - can you put them into a helper func?

Reply:
Done